controllers/provisionserver_controller.go

```diff
@@ -285,6 +285,12 @@ func (r *ProvisionServerReconciler) getProvisionServerProvisioningIP(instance *o
 	pod := podList.Items[0]
 	r.Log.Info(fmt.Sprintf("Found pod %s on node %s", pod.Name, pod.Spec.NodeName))
 
+	// If the pod is not running yet, we don't actually have the server available
+	// on the provisioning IP, so just stop here for now
+	if pod.Status.Phase != corev1.PodRunning {
+		return "", nil
```

Comment:
Do you want to Requeue here instead?

So this perhaps:
return ctrl.Result{RequeueAfter: time.Second * 30}, err